hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -718,6 +718,11 @@ public void processPartitionRuntimeState(PartitionRuntimeState partitionState) {
         lock.lock();
         try {
             final Address sender = partitionState.getEndpoint();
+            if (!node.getNodeExtension().isStartCompleted()) {
```

Comment:
It'll be called when master publishes partition table and that happens after every migration and also periodically (15 sec by default).

That means, it will eventually succeed. 

Normally, this doesn't happen without hot-restart. Because when a node joins to the cluster, it will be ready to receive the partition table.